python/ray/ml/preprocessor.py

```diff
@@ -29,31 +25,63 @@ class Preprocessor(abc.ABC):
     fitting, and uses these attributes to implement its normalization transform.
     """
 
+    class FitStatus(str, Enum):
+        """The fit status of preprocessor."""
+
+        NOT_FITTABLE = "NOT_FITTABLE"
+        NOT_FITTED = "NOT_FITTED"
+        # Only meaningful for Chain preprocessors
+        PARTIALLY_FITTED = "PARTIAL_FITTED"
+        FITTED = "FITTED"
+
     # Preprocessors that do not need to be fitted must override this.
     _is_fittable = True
 
+    def fit_status(self) -> "Preprocessor.FitStatus":
+        if not self._is_fittable:
+            return Preprocessor.FitStatus.NOT_FITTABLE
+        elif self._check_is_fitted():
+            return Preprocessor.FitStatus.FITTED
+        else:
+            return Preprocessor.FitStatus.NOT_FITTED
+
     def fit(self, dataset: Dataset) -> "Preprocessor":
         """Fit this Preprocessor to the Dataset.
 
         Fitted state attributes will be directly set in the Preprocessor.
 
+        Calling it more than once on a preprocessor will overwrite all
+        previously fitted states.
+
         Args:
             dataset: Input dataset.
 
         Returns:
             Preprocessor: The fitted Preprocessor with state attributes.
         """
-        if self.check_is_fitted():
-            raise PreprocessorAlreadyFittedException(
-                "`fit` cannot be called multiple times. "
-                "Create a new Preprocessor to fit a new Dataset."
+        fit_status = self.fit_status()
+        if fit_status == Preprocessor.FitStatus.NOT_FITTABLE:
+            # Just return. This makes Chain Preprocessor easier.
```

Comment:
Don't need to be so specific to Chain here.
```suggestion
            # No-op because there is no state to be fitted.
```